manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -115,8 +110,10 @@ func New(pg plugingetter.PluginGetter, netConfig *NetworkConfig) (networkallocat
 		return nil, err
 	}
 
-	if err := initializeDrivers(reg); err != nil {
-		return nil, err
+	for _, init := range initializers {
+		if err := init(reg, nil); err != nil {
```

Comment:
- https://github.com/moby/moby/pull/45881

Reply:
> The driver config is always nil?! 

Hm... good one, I have to double check if there's a driver I missed that still consumes a config, but at least _most of them_ were not using it, hence me opening;

- https://github.com/moby/moby/pull/45855


>  I wish I had checked before refactoring libnetwork and used a different signature for those Register functions. It's not too late yet to fix my past mistake!

Don't blame yourself! There's so much indirection, and so much "oh, let's add some options, just in case, someday, maybe we'll use them".

I started to chip away more bits in https://github.com/moby/moby/pull/45870, and some of those changes may become obsolete if we carve out larger bits, but it was in an attempt to get clearer what's actually used.